remotespark/livyclientlib/sqlquery.py

```diff
@@ -38,13 +44,23 @@ def to_command(self, kind):
             raise ValueError("Kind '{}' is not supported.".format(kind))
 
     def execute(self, session):
-        (success, records_text) = self._get_records(session)
-        if not success:
-            raise DataFrameParseException(records_text)
-        return self._records_to_dataframe(records_text)
-
-    def _get_records(self, session):
-        return self.to_command(session.kind).execute(session)
+        self._spark_events.emit_sql_execution_start_event(session.guid, session.kind, session.id, self.guid)
```

Comment:
What about including sampling method and other parameters? I think that would be useful.